packages/ember-metal/lib/property_events.js

```diff
@@ -86,8 +86,7 @@ function notifyPropertyChange(obj, keyName, _meta) {
 
   let possibleDesc = descriptorFor(obj, keyName, meta);
 
-  // shouldn't this mean that we're watching this key?
-  if (possibleDesc !== undefined && possibleDesc.didChange) {
+  if (possibleDesc !== undefined && 'didChange' in possibleDesc) {
```

Comment:
this change seems unrelated and I'm curious why?

Reply:
just to avoid `boolean coercion`

Reply:
Last time I checked `in` was worse than boolean coercion in v8 (cc @bmeurer ) it is better to do possibleDesc.didChange !== undefined

Reply:
I'd go with `possibleDesc.didChange !== undefined`. The `in` operator is not backed by ICs so it only pays off if the objects have vastly different shapes all the time.

Reply:
Thanks for the quick response